DB/DataObject.php

```diff
@@ -4278,7 +4304,30 @@ function free()
     function _get_table() { return $this->table(); }
     function _get_keys()  { return $this->keys();  }
 
-
+  /**
+   * Get a string to append to the query log depending on time taken.
+   *
+   * This is to allow easier grepping for slow queries.
+   *
+   * @param float $timeTaken
+   *
+   * @return string
+   */
+  public function getAlertLevel($timeTaken) {
+    if ($timeTaken >= 20) {
+      return '****';
+    }
+    if ($timeTaken > 10) {
+      return '***';
+    }
+    if ($timeTaken > 5) {
+      return '**';
+    }
+    if ($timeTaken > 1) {
```

Comment:
It's neat how you could search for "as-slow-or-slower-than" (e.g. searching for `**` will match `**` or `***`; similarly, `~~` matches `~~` or `~~~`), although it is a little cryptic. Maybe something more textual would do:

```
001 sec (slow)
005 sec (very-slow)
010 sec (very-very-slow)
020 sec (very-very-very-slow)
```

But I'm just bikeshedding.